### PR TITLE
Add method signature information to Scout API Doc model

### PR DESCRIPTION
**Pull request by brunokoeferli**
301209

### CODE REVIEW

File: org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/doc/ApiDocGenerator.java [hunk 1]
```diff
@@ -192,7 +195,7 @@ protected MethodDescriptor toMethodDescriptor(Method m) {
         .withMethod(m)
         .withHttpMethod(httpMethod)
         .withPath(path)
-        .withSignature(signature)
+        .withMethodSignature(methodSignature)
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/doc/ApiDocGenerator.java]: According to other with-methods I wouldn't change this method name -> withSignature(methodSignature)

File: org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/doc/ApiDocGenerator.java [hunk 2]
```diff
@@ -864,6 +873,169 @@ public MethodDescriptor withProduces(String produces) {
     }
   }
 
+  public static class MethodSignatureDescriptor implements Comparable<MethodSignatureDescriptor> {
+
+    private Method m_method;
+
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/doc/ApiDocGenerator.java:890]: remove newline

File: org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/doc/ApiDocGenerator.java [hunk 3]
```diff
@@ -864,6 +873,169 @@ public MethodDescriptor withProduces(String produces) {
...
+              if (genericType instanceof Class<?>) {
+                return (Class<?>) genericType;
+              }
+              // fallback for unknown generic types (like java.lang.reflect.WildcardType)
+              return Object.class;
+            })
+            .collect(Collectors.toList());
+      }
+      else {
+        throw new UnsupportedOperationException();
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/doc/ApiDocGenerator.java]: add a few context infos, e.g. "unsupported type " + type

**Review by paolobazzi — APPROVED** [review]
Looks good, see comments for some minor issues

File: org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/doc/ApiDocGenerator.java (hunk 3 above)
Comment (brunokoeferli) [reply org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/doc/ApiDocGenerator.java]: good point, you are absolutely right

File: org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/doc/ApiDocGenerator.java (hunk 2 above)
Comment (brunokoeferli) [reply org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/doc/ApiDocGenerator.java:890]: theoretically correct, but i based the formatting on the other descriptor classes. e.g. see MethodDescriptor.m_method

File: org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/doc/ApiDocGenerator.java (hunk 1 above)
Comment (brunokoeferli) [reply org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/doc/ApiDocGenerator.java]: i will change it.

File: org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/doc/ApiDocGenerator.java (hunk 2 above)
Comment (paolobazzi) [reply org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/doc/ApiDocGenerator.java:890]: ok

**Pull request merged**